realscale.py

```diff
@@ -59,16 +82,83 @@ def calc_scale_center(self, p1x, p1y, p2x, p2y):
         if line_x < pagecenter_x:
             # Left hand side
             if line_y < pagecenter_y:
-                scale_center = [minx,miny] # UL
+                scale_center = (minx,miny) # UL
             else:
-                scale_center = [minx,maxy] # LL
+                scale_center = (minx,maxy) # LL
         else: # Right hand side
             if line_y < pagecenter_y:
-                scale_center = [maxx,miny] # UR
+                scale_center = (maxx,miny) # UR
             else:
-                scale_center = [maxx,maxy] # LR
+                scale_center = (maxx,maxy) # LR
         #inkex.debug("%s  %s,%s" % (scale_center, pagecenter_x*2, pagecenter_y*2))
         return scale_center
+    
+    def create_ruler(self, parent, width, pos, value, drawnscale):
+        """ Draw Scale rule
+            - Position above user's straightline.
+            - Ruler shows two units together. First one cut into 5
+            pos is a tuple e.g. (0,0)
+            Todo:
+            - Add magnification e.g. 2:1 for small drawings
+            - Override scale if result is > page size
```

Comment:
yes - just a todo :) I will remove it.
I imagined it might be an "auto" option to do it all for them.